go/vt/vtgate/evalengine/fn_time.go

```diff
@@ -1249,6 +1254,53 @@ func (call *builtinLastDay) compile(c *compiler) (ctype, error) {
 	return ctype{Type: sqltypes.Date, Flag: arg.Flag | flagNullable}, nil
 }
 
+func fromDays(loc *time.Location, d int64) *datetime.Date {
+	// 3652424 days corresponds to maximum date i.e. 9999-12-31
+	// mysql returns 0000-00-00 for days below 366
+	if d > 3652424 || d < 366 {
```

Comment:
mysql returns `0000-00-00` for `d < 366` and `d >= (3652424+76)`, but `NULL` for `d > 3652424 && d < (3652424+76)`. As it is not possible to create a `datetime.Date{0, 0, 0}`. So, returned `nil` here. Is there any other approach we should consider here?

Also, mysql docs says that It is not intended for use with values that precede the advent of the Gregorian calendar (1582). So, should we a check for that too?